master/internal/api_experiment.go

```diff
@@ -1583,15 +1584,38 @@ func (a *apiServer) MoveExperiment(
 	if err != nil {
 		return nil, err
 	}
-	p, err := a.GetProjectByID(req.DestinationProjectId, *curUser)
+
+	// check that user can view destination project
+	destProject, err := a.GetProjectByID(req.DestinationProjectId, *curUser)
 	if err != nil {
 		return nil, err
 	}
-	if p.Archived {
+	if destProject.Archived {
 		return nil, errors.Errorf("project (%v) is archived and cannot add new experiments.",
 			req.DestinationProjectId)
 	}
 
+	// get experiment info
+	exp, err := a.getExperiment(int(req.ExperimentId))
+	if err != nil {
+		return nil, err
+	}
+
+	// check that user can view source project
+	srcProject, err := a.GetProjectByID(exp.ProjectId, *curUser)
+	if err != nil {
+		return nil, err
+	}
+	if srcProject.Archived {
+		return nil, errors.Errorf("project (%v) is archived and cannot have experiments moved from it.",
+			srcProject.Id)
+	}
+
+	if err = project.AuthZProvider.Get().CanMoveProjectExperiments(*curUser, srcProject,
```

Comment:
nit: can we handle the check experiment, check project logic inside of `CanMoveProjectExperiments`

Reply:
I had the same question when I was looking at authz, so I based it on `CanMoveProject`, where all checks except happen in api_project.go except for the admin/user-id